dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/EventHookControllerTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2004-2022, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller;
+
+import static org.hisp.dhis.web.WebClient.Body;
+import static org.hisp.dhis.web.WebClientUtils.assertStatus;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.hisp.dhis.eventhook.targets.WebhookTarget;
+import org.hisp.dhis.eventhook.targets.auth.ApiTokenAuth;
+import org.hisp.dhis.eventhook.targets.auth.HttpBasicAuth;
+import org.hisp.dhis.jsontree.JsonList;
+import org.hisp.dhis.jsontree.JsonObject;
+import org.hisp.dhis.web.HttpStatus;
+import org.hisp.dhis.webapi.DhisControllerIntegrationTest;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests the {@link EventHookController} using (mocked) REST requests.
+ *
+ * @author Morten Olav Hansen
+ */
+class EventHookControllerTest extends DhisControllerIntegrationTest
+{
+    @Test
+    void testGetEventHooks()
+    {
+        JsonObject eventHooks = GET( "/eventHooks" ).content( HttpStatus.OK );
+        assertTrue( eventHooks.isObject() );
+        assertTrue( eventHooks.has( "eventHooks" ) );
+        assertTrue( eventHooks.get( "eventHooks" ).isArray() );
+        assertTrue( eventHooks.get( "eventHooks" ).asList( JsonObject.class ).isEmpty() );
```

Comment:
For a empty check you can just do `getArray( "eventHooks" ).isEmpty()`.

You do not need the `assertTrue( eventHooks.has( "eventHooks" ) );` because `eventHooks.get( "eventHooks" ).isArray()` implies that.